katdal/h5datav3.py

```diff
@@ -309,13 +309,21 @@ def register_telstate_sensor(name, obj):
         # Obtain flag descriptions from file or recreate default flag description table
         self._flags_description = data_group['flags_description'] if 'flags_description' in data_group else \
             np.array(zip(FLAG_NAMES, FLAG_DESCRIPTIONS))
+        self._flags_select = np.array([0], dtype=np.uint8)
+        self._flags_keep = 'all'
 
         # ------ Extract weights ------
 
-        # check if weight group present, else use dummy weight data
+        # Check if weights and weights_channel datasets are present, else use dummy weight data
         self._weights = data_group['weights'] if 'weights' in data_group else \
-            dummy_dataset('dummy_weights', shape=self._vis.shape[:-1] + (1,), dtype=np.float32, value=1.0)
-        self._weights_description = np.array(zip(WEIGHT_NAMES, WEIGHT_DESCRIPTIONS))
+            dummy_dataset('dummy_weights', shape=self._vis.shape[:-1], dtype=np.float32, value=1.0)
```

Comment:
To make it behave exactly like the real dataset, it should be `np.uint8`. The difference might not be observable though.